odoo/addons/base/models/ir_model.py

```diff
@@ -1961,16 +1955,17 @@ def _update_xmlids(self, data_list, update=False):
             prefix, suffix = data['xml_id'].split('.', 1)
             record = data['record']
             noupdate = bool(data.get('noupdate'))
-            # First create XML ids for parent records, then create XML id for
-            # record. The order reflects their actual creation order. This order
-            # is relevant for the uninstallation process: the record must be
+            # Create xids for implicitly created parents, then create xid for
+            # record.
+            #
+            # The order reflects their actual creation order. This order is
+            # relevant for the uninstallation process: the record must be
             # deleted before its parent records.
-            for parent_model, parent_field in record._inherits.items():
+            for parent_field in data.get('create_parent', []):
                 parent = record[parent_field]
-                puffix = suffix + '_' + parent_model.replace('.', '_')
+                puffix = suffix + '_' + parent._table
```

Comment:
I don't think it matters, the suffix is just a way to generate the parent xid unambiguously from the child. Since this should be the only place where we do that after this change it shouldn't matter.

Reply:
It is a non-necessary breakage risk.